docs/develop/go/converters-and-encryption.mdx

```diff
@@ -118,12 +119,22 @@ c, err := client.Dial(client.Options{
 //...
 ```
 
-For reference, see the following samples:
+- Data **encoding** is performed by the client using the default converter provided by Temporal or your custom Data Converter when passing input to the Temporal Cluster. For example, plain text input is usually serialized into a JSON object, and can then be compressed or encrypted.
```

Comment:
We need to be careful about the terms. We can easily get confused.

This is taken out of an explanation I gave on Slack a few days ago:

> "Data Converters" is the high-level idea. The key concept here is that there are three major components to Data Converters:
> 
> **Payload Converters**: they handle serialization and deserialization of application objects; i.e. convert from JS objects to bytes in a Buffer. In TS, the default is to use JSON.stringify() and JSON.parse() which is normally enough.
> 
> **Failure Converters**: they play a role kind of similar to **Payload Converters**, but specifcially for Error/Failures objects. Let’s forget about this one for now.
> 
> **Payload Codecs**: they transform the byte representation of a payload, in and out; i.e. they convert a Buffer into another Buffer.
> 
> Now, Payload Converters and Failure Converters operate inside the context of the Workflow sandbox.
> 

This section is about Payload Codecs, but here, we're mixing in ideas that belong to Payload Converters.

But Payload Codecs operate outside of the sandbox, on your main Node’s thread.


Encryption and compression should be implemented as Payload Codecs, not Payload Converters.

Reply:
Actually, I think it would be desirable to give a quick top-down overview of Data Converters and its three components — Payload Converters/Failure Converters/Payload Codecs — very early on this page, before getting into code. The Data Converters setup code may look horribly complex if you don't understand this simple concept first.